src/cpp/utils.h

```diff
@@ -52,13 +54,40 @@ class ScopeGuard {
     std::function<void()> f_;
 };
 
-} // effil
+class Logger
```

Comment:
Can you please create separate files `log.h` and `log.cpp`.
This feature is quite have to be part of utils.